qiskit/dagcircuit/dagcircuit.py

```diff
@@ -861,19 +863,89 @@ def idle_wires(self, ignore=None):
                 else:
                     yield wire
 
-    def size(self):
-        """Return the number of operations."""
-        return len(self._multi_graph) - 2 * len(self._wires)
+    def size(self, *, recurse: bool = False):
+        """Return the number of operations.  If there is control flow present, this count may only
+        be an estimate, as the complete control-flow path cannot be staticly known.
+
+        Args:
+            recurse: if ``True``, then recurse into control-flow operations.  For loops with
+                known-length iterators are counted unrolled.  If-else blocks take the longer case of
+                the two branches.  While loops are counted as if the condition runs once only.
```

Comment:
I'm slightly on the fence about `If-Else` blocks. If the intent is for `recurse=True` to be a minimization target for an optimization loop, would the sum of the `True` and `False` branches be better (to avoid a case where the transpiler would miss room for further optimization on the already-shorter branch)?

Reply:
Yeah, that makes more sense to me.  Originally I think I was still trying to make a `size` number that "made sense" (in this case, "how many instructions are likely to be run?"), but per the rest of the discussion, making it make sense for an optimisation target is probably better.

Reply:
Changed in d0d8f59a0.